### PR TITLE
Returning company information

### PR DESCRIPTION
**Pull request by s-e-v**
The "userIdentifier" function returns the company information based on the user's token post login.

The tests check for various scenarios (the function does not check if the token is not available as the user would be unable to be on the page in the first place).

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -61,6 +61,17 @@ class ANDGatewayClient {
     });
   }
 
+  userIdentifier() {
```
Comment (oskanberg) [inline src/index.js]: I think this is a bad name. How about `getUserCompany()`?

File: src/index.js [hunk 2]
```diff
@@ -61,6 +61,17 @@ class ANDGatewayClient {
...
+    try {
+      const token = Cookies.get(this._envConfig.SSO_COOKIE_NAME);
```
Comment (oskanberg) [inline src/index.js]: This should call `this._getTokenCookie()`

File: src/index.js [hunk 3]
```diff
@@ -61,6 +61,17 @@ class ANDGatewayClient {
...
+      const payload = decodeJWT(token);
+      const company = payload["http://sso.and.digital/company"];
+      return company;
+    } catch (error) {
+      return true;
```
Comment (oskanberg) [inline src/index.js]: Is `true` a good default value for company?

File: test/unit.js [hunk 4]
```diff
@@ -127,3 +127,32 @@ describe("getExpiryDate", () => {
...
+    client._setTokenCookie(token);
+    expect(client.userIdentifier()).toBe(undefined);
+  });
+
+  it("should return undefined if token is empty", () => {
+    const token = "";
+    client._setTokenCookie(token);
+    expect(client.userIdentifier()).toBe(undefined);
```
Comment (oskanberg) [inline test/unit.js]: Really? As it is, doesn't it return `true`?

File: src/index.js (hunk 1 above)
Comment (s-e-v) [reply src/index.js]: Good idea, implemented.

File: src/index.js (hunk 2 above)
Comment (s-e-v) [reply src/index.js]: Now it works again, fixed.

File: src/index.js (hunk 3 above)
Comment (s-e-v) [reply src/index.js]: Fixed, please check the new code.

File: test/unit.js (hunk 4 above)
Comment (s-e-v) [reply test/unit.js]: Fixed, please check the new code.

**Comment by s-e-v** [comment]
Please check the new code/tests:
1) Renamed the function
2) Removed the promise (unnecessary)
3) 2 IFs to check at different stages if the token is invalid, and if the company key exists in the token, otherwise return the company info
4) Tests updated to match the new changes

File: src/index.js [hunk 5]
```diff
@@ -61,6 +61,21 @@ class ANDGatewayClient {
     });
   }
 
+  getUserCompany() {
+    const token = this._getTokenCookie();
+    if (token == undefined) {
+      return "No token found/Token is not valid";
```
Comment (oskanberg) [inline src/index.js]: how about `throw new Error('Authentication token not found')`

File: src/index.js [hunk 6]
```diff
@@ -61,6 +61,21 @@ class ANDGatewayClient {
...
+    } else {
+      const payload = decodeJWT(token);
+      const company = payload["http://sso.and.digital/company"];
+      if (company == undefined) {
+        return "The value does not exist in the token";
```
Comment (oskanberg) [inline src/index.js]: I think that returning undefined or '' works here

File: src/index.js [hunk 7]
```diff
@@ -61,6 +61,21 @@ class ANDGatewayClient {
     });
   }
 
+  getUserCompany() {
```
Comment (oskanberg) [inline src/index.js]: Please add JSDoc comment (like the other methods) and rebuild - this will automatically add it to the readme.

**Pull request merged**